app/src/main/java/lt/vilnius/tvarkau/NewProblemActivity.java

```diff
@@ -211,26 +211,25 @@ public void sendProblem() {
                 photos = new String[problemImagesURIs.size()];
 
                 photoObservable = Observable.from(problemImagesURIs)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(Schedulers.io())
                     .map(uri -> Uri.fromFile(new File(uri.toString())))
                     .map(uri -> {
-                        Bitmap bitmap = null;
-                        try {
-                            bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), uri);
-                            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-                            bitmap.compress(Bitmap.CompressFormat.JPEG, 60, byteArrayOutputStream);
-                            byte[] byteArrayImage = byteArrayOutputStream.toByteArray();
-                            return Base64.encodeToString(byteArrayImage, Base64.NO_WRAP);
-                        } catch (IOException e) {
-                            e.printStackTrace();
-                            FirebaseCrash.report(e);
-                            return null;
-                        }
+                        Bitmap bitmap = BitmapUtils.createBitmap(uri);
+                        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+                        bitmap.compress(Bitmap.CompressFormat.JPEG, 70, byteArrayOutputStream);
+                        byte[] byteArrayImage = byteArrayOutputStream.toByteArray();
+                        return Base64.encodeToString(byteArrayImage, Base64.NO_WRAP);
```

Comment:
That bitmap compression/conversion code I would suggest to move to `BitmapUtils` class too.